lib/active_model/serializer/array_serializer.rb

```diff
@@ -11,9 +11,8 @@ def initialize(resources, options = {})
         @root = options[:root]
         @object = resources
         @serializers = resources.map do |resource|
-          serializer_class = options.fetch(:serializer) do
-            ActiveModel::Serializer.serializer_for(resource)
-          end
+          serializer_context_class = options.fetch(:serializer_context_class, ActiveModel::Serializer)
+          serializer_class = options.fetch(:serializer) { serializer_context_class.serializer_for(resource) }
```

Comment:
The previous line(`serializer_context_class = options.fetch(:serializer_context_class, ActiveModel::Serializer)`) ensures it is not required. Am I misunderstanding?